media/server/main/source/MediaPipelineServerInternal.cpp

```diff
@@ -587,12 +587,26 @@ bool MediaPipelineServerInternal::haveDataInternal(MediaSourceStatus status, uin
         return true;
     }
     m_activeRequests->erase(needDataRequestId);
+
+    int &counter = m_attachedSourcesForCounter[mediaSourceType];
+
     if (status != MediaSourceStatus::OK && status != MediaSourceStatus::EOS)
     {
-        RIALTO_SERVER_LOG_WARN("Data request for needDataRequestId: %u received with wrong status", needDataRequestId);
+        counter++;
+        if (counter == 10)
+        {
+            RIALTO_SERVER_LOG_WARN("Data request for needDataRequestId: %u received with wrong status",
```

Comment:
I think that you can add info that NO_AVAILABLE_SAMPLES was received in X consecutive responses to this log.
Also, when counter != X, you can still print it, but with DEBUG level.